### DISCUSSION TITLE
Does virtio-vsock support udp now?

### DISCUSSION THREAD

**Discussion by xuehuihui** [virtio-win/kvm-guest-drivers-windows#871] [Q&A] [answered]
Is it possible to support it in the future?

The following code creates socket that returns the invalid socket
`    SOCKET sSocket = INVALID_SOCKET;
    if (g_channel == VSock) {
        ADDRESS_FAMILY family = ViosockGetAF();
        sSocket = socket(family, SOCK_DGRAM, 0);
    }
`
@YanVugenfirer

**Comment by YanVugenfirer** [comment, marked as answer]
@xuehuihui virtio-vsock on Windows is a work in progress.  It looks that now it doesn't support "SOCK_DGRAM" only "SOCK_STREAM":
https://github.com/virtio-win/kvm-guest-drivers-windows/blob/42e33f344b2e702ef3a2fbc8c8844e2aff305b39/viosock/sys/Socket.c#L2005

**Comment by YanVugenfirer** [comment]
I suggest opening an issue for tracking.
